libs/groupSubscriptionManager.js

```diff
@@ -68,8 +71,7 @@ function chargeGroup (group, callback)
     // TODO should expire only in case of failed payment
     // otherwise retry
     if (response.AuthorizationDetails.AuthorizationStatus.State === 'Declined') {
-      cancelSubscription(group);
-      return;
+      return cancelSubscription(group);
```

Comment:
Hmm not exactly... it executes `cancelSubscription ` but if cancel subscription never errors it'll keep executing like nothing happened. In a promise you should `throw` an error and then handle it (cancelling the subscription) in a `.catch` block